pkgs/by-name/az/azure-pipelines-agent/package.nix

```
@@ -0,0 +1,262 @@
+  LOCALE_ARCHIVE = lib.optionalString (
+    !DOTNET_SYSTEM_GLOBALIZATION_INVARIANT
+  ) "${glibcLocales}/lib/locale/locale-archive";
+
+  patches = [
+    ./patches/dont-install-service.patch
+    ./patches/env-sh-use-runner-root.patch
+    ./patches/fix-logging-cs-create-folder.patch
+    ./patches/host-context-dirs.patch
+    ./patches/no-wrap-libicu.patch
+    ./patches/use-get-directory-for-diag.patch
+    ./patches/fix-license-dir.patch
+  ];
+
+  preConfigure = ''
+    # Generate src/Microsoft.VisualStudio.Services.Agent/BuildConstants.cs
+    dotnet msbuild \
+      -t:GenerateConstant \
+      -p:ContinuousIntegrationBuild=true \
+      -p:Deterministic=true \
+      -p:PackageRuntime="${dotnetCorePackages.systemToDotnetRid stdenv.hostPlatform.system}" \
+      -p:AgentVersion="${version}" \
+      src/dir.proj
+  '';
+
+  nativeBuildInputs = [
+    which
+    gitMinimal
+    openssl
+    krb5
+    icu
+    zlib
+    coreutils
+  ];
+
+  buildInputs = [
+    (lib.getLib stdenv.cc.cc)
+    bash
+  ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  dotnetFlags = [
+    "-p:PackageRuntime=${dotnetCorePackages.systemToDotnetRid stdenv.hostPlatform.system}"
+    "-p:TargetFramework=net8.0"
+  ];
+
+  projectFile = [
+    "src/Agent.Sdk/Agent.Sdk.csproj"
+    "src/Agent.Listener/Agent.Listener.csproj"
+    "src/Microsoft.VisualStudio.Services.Agent/Microsoft.VisualStudio.Services.Agent.csproj"
+    "src/Agent.Worker/Agent.Worker.csproj"
+    "src/Agent.PluginHost/Agent.PluginHost.csproj"
+    "src/Agent.Plugins/Agent.Plugins.csproj"
+  ];
+  nugetDeps = ./deps.json;
+
+  preCheck = ''
+    mkdir -p _layout/externals
+    ln -s ${nodejs_20} _layout/externals/node20_1
+  '';
+
+  postInstall =
+    ''
+      mkdir -p $out/bin
+      mkdir -p $out/lib/${pname}/en-US
+
+      install -m755 src/Misc/layoutbin/AgentService.js          $out/lib/${pname}
+      install -m755 src/Misc/layoutbin/en-US/strings.json       $out/lib/${pname}/en-US
+      install -m755 src/Misc/layoutbin/runsvc.sh                $out/lib/${pname}
+      install -m755 src/Misc/layoutroot/config.sh               $out/lib/${pname}
+      install -m755 src/Misc/layoutroot/env.sh                  $out/lib/${pname}
+      install -m755 src/Misc/layoutroot/license.html            $out/lib/${pname}
+      install -m755 src/Misc/layoutroot/reauth.sh               $out/lib/${pname}
+      install -m755 src/Misc/layoutroot/run.sh                  $out/lib/${pname}
+
+      # env.sh is patched to not require any wrapping
+      ln -sr "$out/lib/${pname}/env.sh" "$out/bin/"
+
+      substituteInPlace $out/lib/${pname}/config.sh \
+          --replace-fail './bin/Agent.Listener' "$out/bin/Agent.Listener"
+
+      substituteInPlace $out/lib/${pname}/run.sh --replace-fail '"$DIR"/bin/' '"$DIR"/'
+
+      substituteInPlace $out/lib/${pname}/config.sh \
+        --replace 'command -v ldd' 'command -v ${glibc.bin}/bin/ldd' \
```

```suggestion
        --replace-fail 'command -v ldd' 'command -v ${glibc.bin}/bin/ldd' \
```
same for similar replacements below, plain `--replace` is deprecated